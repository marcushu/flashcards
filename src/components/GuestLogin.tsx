interface GuestLoginProps {
  loginHandler: (input: string) => void
}

const GuestLogin = ( { loginHandler }: GuestLoginProps) => {
  return (
    <div className="card">
      <div>
        <button className="loginBtn" onClick={() => loginHandler("GUEST")}>
          Continue as guest
        </button>
        <br /><br />
          and choose from available topics.
      </div>
    </div>
  )
}

export default GuestLogin