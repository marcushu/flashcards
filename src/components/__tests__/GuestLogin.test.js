import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import GuestLogin from '../GuestLogin';

describe("GuestLogin", () => {
    const mockFunk = jest.fn();

    it("renders", () => {
        const { asFragment } = render(
            <GuestLogin loginHandler={mockFunk} />
        );

        expect(asFragment()).toMatchSnapshot()
    });

    it("calls the login function with 'GUEST'", () => {
        const { getByRole } = render(
            <GuestLogin loginHandler={mockFunk} />
        );

        fireEvent.click(getByRole('button', {  name: /continue as guest/i}));

        expect(mockFunk).toHaveBeenCalledWith("GUEST");
    })
});