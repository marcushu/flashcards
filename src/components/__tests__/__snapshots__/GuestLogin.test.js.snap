// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestLogin renders 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div>
      <button
        class="loginBtn"
      >
        Continue as guest
      </button>
      <br />
      <br />
      and choose from available topics.
    </div>
  </div>
</DocumentFragment>
`;
